cloud/ec2_util.go

```diff
@@ -747,3 +747,14 @@ func isEC2InstanceNotFound(err error) bool {
 	}
 	return false
 }
+
+func shouldAssignPublicIPv4Address(h *host.Host, ec2Settings *EC2ProviderSettings) bool {
+	if h.UserHost || h.SpawnOptions.SpawnedByTask {
+		// Spawn hosts and host.create hosts need to have a public IPv4 address
+		// because SSH is currently the only means for the user/task to access
+		// the host.
+		return true
+	}
+
+	return !ec2Settings.DoNotAssignPublicIPv4Address && !ec2Settings.IPv6
```

Comment:
Just out of curiosity, when does this return true? In my mental model, this is reached for task hosts so why wouldn't  they consistently return true/false (like spawn hosts and host.create hosts?)

Reply:
This will return true if neither of these distro settings are set to true. It's not required that either of these be set since they're opt-in, so the default becomes "assign public IPv4 address, don't use IPv6". If it's using IPv6, the old (and new) logic always refused to assign a public IPv4 address, so I just preserved that behavior.

On a side note, I'm pretty sure IPv6 is a dead setting since there's no way to set it from the UI. I imagine it's some leftover from when someone was testing IPv6 long ago, but I kept it working the same with the thought that maybe one day we'll use IPv6.